src/DankCord/Objects.py

```diff
@@ -35,62 +37,112 @@ def __init__(self, data: dict) -> None:
-        self.username: str = data["username"]
-        self.id: int = int(data["id"])
-        self.discriminator: int = int(data["discriminator"])
-        self.bot: bool = data["bot"]
+        self.name: str = data.get("name", "")
+        self.icon_url = data.get("icon_url", "")
 
 
 class ActionRow:
-    def __init__(self, data: dict):
+    def __init__(self, data: dict, message_id: str, client):
         self.components = [
-            Button(i) if i["type"] == 2 else Dropdown(i) for i in data["components"]
+            Button(i, message_id, client) if i["type"] == 2 else Dropdown(i) for i in data["components"]
         ]
 
 
 class DropdownOption:
     def __init__(self, data: dict) -> None:
         self.label: str = data["label"]
         self.default: bool = data.get("default", False)
+        self.value: str = data.get("value", "")
 
 
 class Dropdown:
     def __init__(self, data: dict) -> None:
         self.type = 3
         self.options = [DropdownOption(child) for child in data["options"]]
-
     # TODO: Make a choose function
 
 
 class Button:
-    def __init__(self, data: dict) -> None:
+    def __init__(self, data: dict, message_id: str, client) -> None:
+        self.client = client
+        self.message_id = message_id
         self.type = 2
-        self.emoji: Optional[Emoji] = Emoji(data["emoji"]) if data["emoji"] else None
+        self.emoji: Optional[Emoji] = Emoji(
+            data["emoji"]) if "emoji" in data.keys() else None
         self.label: str = data.get("label", "")
-
         self.disabled: bool = data.get("disabled", False)
         self.custom_id: str = data.get("custom_id", "")
 
-    # TODO: Make a click function
+    def click(self, retry_attempts: int=3):
+        nonce = self.client._create_nonce()
+        data = {
+            "type": 3,
+            "nonce": nonce,
+            "guild_id": self.client.guild_id,
+            "session_id": self.client.session_id,
+            "channel_id": self.client.channel_id,
+            "message_flags": 0,
+            "message_id": self.message_id,
+            "application_id": "270904126974590976",
+            "data": {
+                "component_type": 2,
+                "custom_id": self.custom_id
+            }
+        }
+        for i in range(retry_attempts):
+            response = Response(
+                requests.post(  # type: ignore
+                    "https://discord.com/api/v9/interactions",
+                    json.dumps(data),
+                    http_headers=self.client._tupalize(
+                        {
+                            "Authorization": self.client.token,
+                            "Content-Type": "application/json",
+                        }
+                    ),
+                )
+            )
+            if isinstance(response.data, dict):
+              retry_after = int(response.data.get("retry_after", 0))
+              time.sleep(retry_after)
+            else:
+              break
```

Comment:
Post click handling has not been done? It should confirm the event by waiting for the nonce to appear in ws_cache

Reply:
added. code isnt the best and could definitely be optimized and i'll make sure that it'll be soon